frontend/src/components/common/form/Check.tsx

```diff
@@ -68,21 +68,23 @@ export const Check: React.FC<CheckProps> = ({
 }) => {
   const {
     values,
+    initialValues,
     setFieldValue,
     setFieldTouched,
     errors,
     touched,
     handleBlur,
   } = useFormikContext();
+  const initialChecked = getIn(initialValues, field);
   const touch = getIn(touched, field);
   const checked = getIn(values, field);
   const error = getIn(errors, field);
   const asElement: any = is || 'input';
   useEffect(() => {
-    if (checked === true || checked === false) {
+    if ((checked === true || checked === false) && initialChecked !== checked) {
```

Comment:
this useeffect shouldn't be triggered on page load just because the initialvalue (and therefore the value) is not undefined.